packages/@react-stately/tree/src/useTreeState.ts

```diff
@@ -53,12 +53,13 @@ export function useTreeState<T extends object>(props: TreeProps<T>): TreeState<T
 
   let tree = useCollection(props, nodes => new TreeCollection(nodes, {expandedKeys}));
 
+  let {focusedKey, setFocusedKey} = selectionState;
   // Reset focused key if that item is deleted from the collection.
   useEffect(() => {
-    if (selectionState.focusedKey != null && !tree.getItem(selectionState.focusedKey)) {
-      selectionState.setFocusedKey(null);
+    if (focusedKey != null && !tree.getItem(focusedKey)) {
+      setFocusedKey(null);
     }
-  }, [tree, selectionState.focusedKey]);
+  }, [tree, focusedKey, setFocusedKey]);
```

Comment:
I'm sure there is a good reason, but why is this logic basically same as what's in useTableState and useListState. It feels very copy-paste? Can't we generalize it and share it?

Reply:
maybe? out of scope for this PR though